cli/src/main/java/com/devonfw/tools/ide/commandlet/EditionSetCommandlet.java

```diff
@@ -52,13 +46,14 @@ public void run() {
     ToolCommandlet commandlet = this.tool.getValue();
     String edition = this.edition.getValue();
 
-    if (this.conf.isTrue()) {
-      commandlet.setEdition(edition, true, EnvironmentVariablesType.CONF);
-    } else if (this.home.isTrue()) {
-      commandlet.setEdition(edition, true, EnvironmentVariablesType.USER);
-    } else if (this.workspace.isTrue()) {
-      commandlet.setEdition(edition, true, EnvironmentVariablesType.WORKSPACE);
+    EnvironmentVariablesType env = this.cfg.getValue();
+    if (env == EnvironmentVariablesType.SETTINGS || env == EnvironmentVariablesType.CONF || env == EnvironmentVariablesType.USER
+        || env == EnvironmentVariablesType.WORKSPACE) {
+      commandlet.setEdition(edition, true, env);
+    } else if (env == EnvironmentVariablesType.RESOLVED || env == EnvironmentVariablesType.SYSTEM) {
+      context.error("Invalid option for --cfg: " + env);
```

Comment:
ah, I did not think about this one. In general it was smart to recycle the existing enum. However, in this case, you should create a new enum with a `EnvironmentVariablesType toType()` method so that auto-completion will not suggest options that do not make any sense.